tests/templates/test_subroutines/test_select.py

```diff
@@ -0,0 +1,277 @@
+            ),
+            (
+                [qml.PauliX(wires=0)] * 10,
+                [1, 2, 3],
+                r"Not enough control wires \(3\) for the desired number of operations \(10\). At least 4 control wires required.",
+            ),
+            (
+                [qml.PauliX(wires="a"), qml.PauliY(wires="b"), qml.PauliZ(wires="c")],
+                [1],
+                r"Not enough control wires \(1\) for the desired number of operations \(3\). At least 2 control wires required.",
+            ),
+        ],
+    )
+    def test_too_many_ops(self, ops, control_wires, msg_match):
+        """Test that error is raised if more ops are requested than can fit in control wires"""
+        with pytest.raises(ValueError, match=msg_match):
+            qml.SELECT(ops, control_wires)
+
+
+def select_rx_circuit(angles):
+    """Circuit that uses SELECT for tests."""
+    qml.SELECT([qml.RX(angles[0], wires=[1]), qml.RY(angles[1], wires=[1])], control_wires=0)
+    return qml.expval(qml.PauliZ(wires=1))
+
+
+def manual_rx_circuit(angles):
+    """Circuit that manually creates SELECT for tests."""
+    qml.ctrl(qml.RX(angles[0], wires=[1]), control=0, control_values=0)
+    qml.ctrl(qml.RY(angles[1], wires=[1]), control=0)
+    return qml.expval(qml.PauliZ(wires=1))
+
+
+class TestInterfaces:
+    """Tests that the template is compatible with all interfaces, including the computation
+    of gradients."""
+
+    @pytest.mark.autograd
+    def test_autograd(self):
+        """Tests the autograd interface."""
+        dev = qml.device("default.qubit", wires=2)
+
+        circuit_default = qml.QNode(manual_rx_circuit, dev)
+        circuit_select = qml.QNode(select_rx_circuit, dev)
+
+        input_default = [0.5, 0.2]
+        input_grad = pnp.array(input_default, requires_grad=True)
+
+        grad_fn = qml.grad(circuit_default)
+        grads = grad_fn(input_grad)
+
+        grad_fn2 = qml.grad(circuit_select)
+        grads2 = grad_fn2(input_grad)
+
+        assert np.allclose(grads, grads2)
+
+    @pytest.mark.tf
+    def test_tf(self):
+        """Tests the tf interface."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", wires=2)
+
+        circuit_default = qml.QNode(manual_rx_circuit, dev)
+        circuit_tf = qml.QNode(select_rx_circuit, dev)
+
+        input_default = [0.5, 0.2]
+        input_tf = tf.Variable(input_default)
+
+        assert np.allclose(
+            qml.matrix(circuit_default)(input_default), qml.matrix(circuit_tf)(input_tf)
+        )
+        assert qml.math.get_interface(qml.matrix(circuit_tf)(input_tf)) == "tensorflow"
+
+        with tf.GradientTape() as tape:
+            res = circuit_default(input_tf)
+        grads = tape.gradient(res, [input_tf])
+
+        with tf.GradientTape() as tape2:
+            res2 = circuit_tf(input_tf)
+        grads2 = tape2.gradient(res2, [input_tf])
+
+        assert np.allclose(grads[0], grads2[0])
```

Comment:
[0] is necessary here?

Reply:
May not be necessary here, but I think it ensures we compare the right things.